lib/req/steps.ex

```diff
@@ -1156,6 +1176,18 @@ defmodule Req.Steps do
 
         assert Req.get!(plug: plug, into: []).body == ["echoecho"]
       end
+
+  You can simulate failure conditions by returning exception structs from your plugs.
+  For network related issues, use `Req.TransportError` exception:
+
+      test "network issues" do
+        plug = fn _conn ->
+          %Req.TransportError{reason: :econnrefused}
+        end
```

Comment:
I like this more than the atom.

Reply:
I think this is better than the atom indeed. You could also have some sort of `Req.Plug.halt(conn, reason)`, if you want to keep the Plug contract, and then you store the reason somewhere in the connection.

Reply:
Oh that's a very good idea, because it allows you to keep the public API flexible!

Reply:
reason being exception struct or atom and friends?

maybe Req.Test.transport_error(conn, reason)?

Reply:
Yeah I think `Req.Test.transport_error(conn, reason)` is the same idea, but it reads better. I also like that it keeps it scoped to `Req.Test`. `reason` would be an atom, so that

```elixir
Req.Test.transport_error(conn, :timeout)
```

essentially stores `conn.private.req_transport_error = %Req.TransportError{reason: :timeout}`, which then you can use to return the error. Thoughts?